models/events.go

```diff
@@ -119,6 +121,19 @@ func decodeCadenceEvents(events flow.BlockEvents) (*CadenceEvents, error) {
 		return nil, fmt.Errorf("EVM block can not be nil if transactions are present, Flow block: %d", events.Height)
 	}
 
+	if e.block != nil && len(e.transactions) > 0 {
```

Comment:
at this point if block is nil should be separate check and return error

Reply:
Currently, we have the assumptions that we can have **at most one** EVM Block, so the block can also be `nil`, and we should check against that.